### PR TITLE
update user-agent to include host machine info

### PR DESCRIPTION
**Pull request by stevengum**
updating the User-Agent header set in the BotFrameworkAdapter. 

Accompanies Microsoft/botbuilder-dotnet#816.

### CODE REVIEW

File: libraries/botbuilder/src/botFrameworkAdapter.ts [hunk 1]
```diff
@@ -62,8 +63,14 @@ export interface InvokeResponse {
...
 
+// Retrieve additional information, i.e., host operating system, host OS release, architecture, Node.js version
+const ARCHITECTURE = os.arch();
+const TYPE = os.type();
+const RELEASE = os.release();
+const NODE_VERSION = process.version;
+
 const pjson: any = require('../package.json');
-const USER_AGENT = 'Microsoft-BotFramework/3.1 (BotBuilder JS/' + pjson.version + ')';
+const USER_AGENT = 'Microsoft-BotFramework/3.1 (BotBuilder JS/' + pjson.version + ') (Node.js,Version=' + NODE_VERSION + '; ' + TYPE + ' ' + RELEASE + '; ' + ARCHITECTURE + ')';
```
Comment (daveta) [inline libraries/botbuilder/src/botFrameworkAdapter.ts]: Spec says "BotBuilder/<SDK Version", ie, BotBuilder/4.0.0

**Comment by stevengum** [comment]
@p-nagpal, @daveta could you sign off on this? The constructed User-Agent header will look like this:

> Microsoft-BotFramework/3.1 BotBuilder/4.0.0-preview1.2 (Node.js,Version=v8.6.0; Windows_NT 10.0.17134; x64)

**Comment by stevengum** [comment]
Cross linking to Microsoft/botbuilder-dotnet#825

**Comment by daveta** [comment]
@stevengum Go ahead and merge - i don't have access..

**Comment by stevengum** [comment]
Fixed the merge conflicts, am currently awaiting the results of the Travis build to verify that the only currently broken tests are in botbuilder-dialogs.

**Comment by stevengum** [comment]
No new tests were broken by this PR, although it looks like the LUIS tests are also failing. The test that added to cover the code added is [passing](https://travis-ci.org/Microsoft/botbuilder-js/builds/416023291#L1422).

**Pull request merged**